bot/src/main/java/com/community/tools/service/NotificationService.java

```diff
@@ -0,0 +1,121 @@
+   * @throws IllegalArgumentException If any Discord user corresponding to the GitHub usernames is
+   *                                  not found.
+   */
+  public void sendPullRequestUpdateNotification(
+      List<UserForTaskStatusDto> userForTaskStatusDtoList) {
+    List<String> gitHubNames = userForTaskStatusDtoList.stream()
+        .map(UserForTaskStatusDto::getGitName).collect(
+            Collectors.toList());
+    Map<String, String> gitHubDiscordNames = discordGithubService.getDiscordGithubUsernames(
+        gitHubNames);
+    for (UserForTaskStatusDto dto : userForTaskStatusDtoList) {
+      String discordName = gitHubDiscordNames.get(dto.getGitName());
+      Optional<User> userOptional = discordService.getUserByName(discordName);
+      sendNotification(userOptional, discordName, dto.getTaskStatuses());
+    }
+  }
+
+  /**
+   * Sends a general notification message to the Discord user associated with the given GitHub
+   * username.
+   *
+   * @param gitHubName The GitHub username for which the notification message is intended.
+   * @param message    The message content to be sent to the Discord user.
+   * @throws IllegalArgumentException If the Discord user corresponding to the GitHub username is
+   *                                  not found.
+   */
+  public void sendNotificationMessage(String gitHubName, String message) {
+    String discordName = discordGithubService.getDiscordName(gitHubName);
+    Optional<User> userOptional = discordService.getUserByName(discordName);
+    if (userOptional.isPresent()) {
+      discordService.sendMessageToConversation(textChannelName,
+          userOptional.get().getAsMention() + ", " + message);
+    } else {
+      throw new IllegalArgumentException(NOT_FOUND_DISCORD_USER_MESSAGE + discordName);
+    }
+  }
+
+  /**
+   * Builds a notification message for a Discord user based on their updated task statuses.
+   *
+   * @param user                  The Discord user for whom the notification message is being
+   *                              built.
+   * @param taskNameAndStatusList A list containing TaskNameAndStatus objects representing the
+   *                              updated task statuses.
+   * @return The built notification message as a String.
+   */
+  private String buildNotification(User user, List<TaskNameAndStatus> taskNameAndStatusList) {
+    StringBuilder announcementText = new StringBuilder();
+    announcementText.append("Hey ").append(user.getAsMention())
+        .append("! Your PR statuses updated on: ");
+    for (TaskNameAndStatus taskNameAndStatus : taskNameAndStatusList) {
+      announcementText.append(System.lineSeparator());
+      announcementText.append(taskNameAndStatus.getTaskName()).append(" :")
+          .append(taskNameAndStatus.getTaskStatus());
+    }
+    return announcementText.toString();
+  }
+
+  private void sendNotification(Optional<User> userOptional, String discordName,
```

Comment:
Is this the old version of method? Or why do we have 2 methods to send notifications